### ISSUE TITLE
[Failing Test] gce-cos-master-serial (ci-kubernetes-e2e-gci-gce-serial)

### ISSUE THREAD

**Issue by droslean** [kubernetes/kubernetes#88441]
**Which jobs are failing**:
`gce-cos-master-serial (ci-kubernetes-e2e-gci-gce-serial)`

**Which test(s) are failing**:
```
[sig-scheduling] SchedulerPreemption [Serial] validates basic preemption works
```
```
[sig-scheduling] SchedulerPreemption [Serial] validates lower priority pod preemption by critical pod
```
```
[sig-scheduling] SchedulerPriorities [Serial] PodTopologySpread Scoring validates pod should be preferably scheduled to node which makes the matching pods more evenly distributed
```
```
[sig-scheduling] SchedulerPredicates [Serial] validates pod overhead is considered along with resource limits of pods that are allowed to run verify pod overhead is accounted for
```

**Testgrid link**:
https://testgrid.k8s.io/sig-release-master-informing#gce-cos-master-serial

**Reason for failure**:
see https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/1231581843564793859

**Anything else we need to know**:
/sig scheduling
/assign @ahg-g 
/cc @kubernetes/ci-signal 
/priority critical-urgent
/milestone v1.18

**Comment by ahg-g** [comment]
/assign @Huang-Wei 

Wei, those are failing tests related to both pod spread and pod overhead.

**Comment by Huang-Wei** [comment]
This issue consists of 3 different symptoms:

- [ ] [sig-scheduling] SchedulerPriorities [Serial] PodTopologySpread Scoring validates pod should be preferably scheduled to node which makes the matching pods more evenly distributed - will be resolved by #88552
- [ ] [sig-scheduling] SchedulerPreemption [Serial]
- [ ] [sig-scheduling] SchedulerPredicates [Serial] validates pod overhead is considered along with resource limits of pods that are allowed to run verify pod overhead is accounted for

I'm working on PRs resolving the 1st and 2nd one. Haven't started on the 3rd one yet, @egernst if you have time, feel free to debug the 3rd one.

**Comment by egernst** [comment]
For #3, @Huang-Wei, I believe we need to check the feature gate status, and skip the test if not enabled.

**Comment by Huang-Wei** [comment]
@egernst Ah, I _thought_ you have already enabled in `kube_features.go`, can you open a PR enable that?

**Comment by egernst** [comment]
I will for 1.18, but waiting on e2e_node test to merge first.

Adding a check is probably a good idea either way -- testing PR now.

**Comment by egernst** [comment]
If easier: here's PR to enable the featuregate: https://github.com/kubernetes/kubernetes/pull/88338

**Comment by Huang-Wei** [comment]
@droslean This issue can be marked as resovled as [4 sub-issues](https://github.com/kubernetes/kubernetes/issues/88441#issuecomment-591141004) have all been fixed.

/close

Feel free to reopen or open a new issue if you spot a new flake.
